### ISSUE TITLE
Error after creating listener

### ISSUE THREAD

**Issue by SamK** [terraform-providers/terraform-provider-openstack#233]
I'm trying to create a lbaas listener with Terraform but `terraform apply` fails with an error

### Terraform Version

```
# terraform -v
Terraform v0.11.3
+ provider.openstack v1.2.0
```
I did not test other versions.

### Affected Resources

One of these two, I'm not really sure:

- "openstack_lb_loadbalancer_v2"
- "openstack_lb_listener_v2"

### Terraform Configuration Files

```hd
provider "openstack" {
}

resource "openstack_lb_loadbalancer_v2" "loadbalancer_1" {
  name = "loadbalancer_1"
  vip_subnet_id = "17460e62-21ae-4878-8638-a9f924bdce75"
}

resource "openstack_lb_listener_v2" "listener_1" {
  name = "listener_1"
  protocol = "HTTP"
  protocol_port = 80
  loadbalancer_id = "${openstack_lb_loadbalancer_v2.loadbalancer_1.id}"
}
```

I swear the `OS_*` environment variables are setup properly.

### Debug Output

https://gist.github.com/SamK/ee38da965679f93abeed1efb02d3b40c

### Expected Behavior

Terraform should not fail after creating the listener.

### Actual Behavior

Terraform cannot unmarshal string into Go struct field LoadBalancer.listeners of type listeners.Listener.
It looks like the provider does not understand the json return by the Neutron endpoint.

### Steps to Reproduce

The creation of the network is not referenced in the Terraform configuration file because it already exists in my project.

1. `terraform apply`

After the first step, `terraform plan` starts failing as well while trying to refresh state.

### Important Factoids

- OpenStack version is Ocata.
- I can create the lbaas without problem using the neutron command line.

**Comment by jtopjian** [comment]
@SamK Thank you for reporting this.

> Terraform cannot unmarshal string into Go struct field LoadBalancer.listeners of type listeners.Listener. It looks like the provider does not understand the json return by the Neutron endpoint.

That definitely looks to be the case, but it's bizarre why... 

Do you know what LB driver/plugin is being used in the OpenStack cloud?

This is the expected output from a vanilla OpenStack Ocata/Pike cloud with the haproxy driver:

```json
{
  "loadbalancer": {
    "admin_state_up": true,
    "description": "",
    "id": "6ffe4e37-7c76-4f29-b747-4bdde6628982",
    "listeners": [
      {
        "id": "18a776f5-0c0e-4f27-8f30-73f53f5c7026"
      }
    ],
    "name": "loadbalancer_1",
    "operating_status": "ONLINE",
    "pools": [],
    "provider": "haproxy",
    "provisioning_status": "ACTIVE",
    "tenant_id": "5aa69d9135524fddab887759f3b6c92d",
    "vip_address": "192.168.199.8",
    "vip_port_id": "9bbe1bd8-f502-4607-bc53-33856f7fa2a1",
    "vip_subnet_id": "bdc5dbd0-6127-4329-9c1f-723d0febf22c"
  }
}
```

Note how the output is slightly different: it's an array/list of objects rather than an array of strings.

**Comment by SamK** [comment]
Hi,
Thank you for your answer.


The output given by my Ocata:

```
{
  "loadbalancer": {
    "admin_state_up": true,
    "description": "",
    "id": "951893bc-d048-45e9-b24d-80772fc65b02",
    "listeners": [
      "f844818c-4e5d-41fc-b8f6-a2100d501d0d",
      "e82919a3-59b4-473d-a248-7b83bc764840"
    ],
    "name": "loadbalancer_1",
    "operating_status": "ONLINE",
    "provisioning_status": "ACTIVE",
    "tenant_id": "006ce30396044fe09cdf049fa3936860",
    "vip_address": "192.168.34.19",
    "vip_port_id": "b13df214-2fa9-4019-b0c0-8c07e5ec9c2e",
    "vip_subnet_id": "17460e62-21ae-4878-8638-a9f924bdce75"
  }
}
```

OpenStack is hosted on Ubuntu provided by the Mirantis Cloud Platform.
The version neutron-lbaas is 10.0.1, packaged in a Ubuntu package (`python-neutron-lbaas` version `2:10.0.1-1~u16.04+mcp2`).

I don't know where I'm supposed to investigate to find out where this change comes from.

**Comment by jtopjian** [comment]
@SamK Thank you for the info! I'll dig into this.

**Comment by SamK** [comment]
Hi,

I digged a bit and found out we are not using Octavia but the OpenContrail plugin version 3.1.1.

What I am experiencing above is a [known bug](https://bugs.launchpad.net/juniperopenstack/+bug/1733107) solved in version 3.2.3.3. The reporter even mentioned Terraform in the bug description.

I guess we will have to update our infrastructure.

I am closing this issue.

Best regards,

Samuel

**Comment by jtopjian** [comment]
@SamK Thank you! I was really scratching my head over this one.
